### PR TITLE
fix(providers): update linkedin oidc issuer

### LINKED ISSUE #8831: Auth provider for Linkedin not working
### Environment

System:
    OS: Linux 6.2 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (12) x64 AMD Ryzen 5 3600X 6-Core Processor
    Memory: 2.63 GB / 15.53 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.4.0 - /usr/local/bin/node
    npm: 9.7.2 - /usr/local/bin/npm

### Reproduction URL

https://github.com/nextauthjs/next-auth-example

### Describe the issue

As per the documentation from Linkedin, I've set up a new LinkedIn app, and added "Sign In with LinkedIn using OpenID Connect" as a product.

At first I had some problems when not specifying scope.

``
    providers: [
        LinkedIn({
            clientId: process.env.LINKEDIN_ID,
            clientSecret: process.env.LINKEDIN_SECRET,
        })
    ], pages: {
        signIn: '/register-cv'
    },
``

This returns an 'unauthorized_scope_error' for r_emailaddress. Managed to fix that issue by providing scopes as per the documentation from Microsoft:

> Authenticating Members
New members logging in to your service for the first time will need to follow the Authenticating with OAuth 2.0 Guide. When requesting the authorization code in Step 2 of the OAuth 2.0 Guide, make sure you use the OpenID scope openid to get the ID Token. We are also introducing new scopes profile and email.

> openid	Required to indicate the application wants to use OIDC to authenticate the member.
> profile	Required to retrieve the member's lite profile including their id, name, and profile picture.
> email	Required to retrieve the member's email address.
> After successful authentication, you will receive the member's access token and ID token.

> If your application does not have these permissions provisioned, you can request access through the Developer Portal. Select your app from My Apps, navigate to the Products tab, and request the Sign in with LinkedIn using OpenID Connect product.

Link: https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin-v2

Updated scopes to match the documentation from Microsoft:

`LinkedIn({
            clientId: process.env.LINKEDIN_ID,
            clientSecret: process.env.LINKEDIN_SECRET,
            authorization: { params: { scope: 'openid profile email' } }})`
            
Now getting this error:

`https://next-auth.js.org/errors#oauth_callback_error unexpected iss value, expected undefined, got: https://www.linkedin.com {
  error: RPError: unexpected iss value, expected undefined, got: https://www.linkedin.com
      at Client.validateJWT (/home/deb/PhpstormProjects/cvmaker/node_modules/openid-client/lib/client.js:931:15)
      at Client.validateIdToken (/home/deb/PhpstormProjects/cvmaker/node_modules/openid-client/lib/client.js:766:60)
      at Client.callback (/home/deb/PhpstormProjects/cvmaker/node_modules/openid-client/lib/client.js:505:18)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async oAuthCallback (/home/deb/PhpstormProjects/cvmaker/node_modules/next-auth/core/lib/oauth/callback.js:109:16)
      at async Object.callback (/home/deb/PhpstormProjects/cvmaker/node_modules/next-auth/core/routes/callback.js:52:11)
      at async AuthHandler (/home/deb/PhpstormProjects/cvmaker/node_modules/next-auth/core/index.js:208:28)
      at async NextAuthApiHandler (/home/deb/PhpstormProjects/cvmaker/node_modules/next-auth/next/index.js:22:19)
      at async NextAuth._args$ (/home/deb/PhpstormProjects/cvmaker/node_modules/next-auth/next/index.js:108:14) {
    name: 'OAuthCallbackError',
    code: undefined
  },
  providerId: 'linkedin',
  message: 'unexpected iss value, expected undefined, got: https://www.linkedin.com'
`

image

### How to reproduce

1: Set up new app on Linkedin, add "Sign In with LinkedIn using OpenID Connect" as a product.
2: Add authentication provider for Linkedin:

```import NextAuth from "next-auth"
import LinkedIn from "next-auth/providers/linkedin"

export const authOptions = {
    secret: 'NOT SO SECRET ANYMORE',
    providers: [
        LinkedIn({
            clientId: process.env.LINKEDIN_ID,
            clientSecret: process.env.LINKEDIN_SECRET,
            authorization: { params: { scope: 'openid profile email' } },
        })
    ], pages: {
        signIn: '/register-cv'
    },
}
export default `NextAuth(authOptions);

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- LinkedIn API changed
- See: https://github.com/nextauthjs/next-auth/issues/8831#issuecomment-2067910984

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

Fixes #8831

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Review by paramaggarwal — APPROVED** [review]

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Can we add `LinkedIn` signin to our deployed examples to make sure it works properly when deployed? :pray:

File: packages/core/src/providers/linkedin.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export default function LinkedIn<P extends LinkedInProfile>(
     name: "LinkedIn",
     type: "oidc",
     client: { token_endpoint_auth_method: "client_secret_post" },
-    issuer: "https://www.linkedin.com",
+    issuer: "https://www.linkedin.com/oauth",
     jwks_endpoint: "https://www.linkedin.com/oauth/openid/jwks",
```
Comment (balazsorban44) [inline packages/core/src/providers/linkedin.ts:82]: we could drop this line, as it is [discovered](https://www.linkedin.com/oauth/.well-known/openid-configuration)

Comment (balazsorban44) [inline packages/core/src/providers/linkedin.ts:82]:
```suggestion
```

We should be able to drop this line, as it is [discovered](https://www.linkedin.com/oauth/.well-known/openid-configuration)

**Comment by ndom91** [comment]
> Can we add `LinkedIn` signin to our deployed examples to make sure it works properly when deployed? 🙏

Yeah, I'll double check the vercel env vars stuff and add it to the open [sveltekit](https://github.com/nextauthjs/next-auth/pull/10673) and [express](https://github.com/nextauthjs/next-auth/pull/10675) PRs as well

File: packages/core/src/providers/linkedin.ts (hunk 1 above)
Comment (ndom91) [reply packages/core/src/providers/linkedin.ts]: Merged :+1:

**Pull request merged**